packages/dropdownable/components/Dropdownable.tsx

```diff
@@ -0,0 +1,251 @@
+import React from "react";
+import styled, { css } from "react-emotion";
+
+import Overlay from "../../shared/components/Overlay";
+import DropdownContents from "./DropdownContents";
+import resizeEventManager from "../../utilities/resizeEventManager";
+
+export enum Direction {
+  BottomLeft = "bottom-left",
+  BottomRight = "bottom-right",
+  TopLeft = "top-left",
+  TopRight = "top-right"
+}
+
+const DEFAULT_DIRECTION_PREFERENCES = [
+  Direction.BottomLeft,
+  Direction.BottomRight,
+  Direction.TopRight,
+  Direction.TopLeft
+];
+
+export interface DropdownableProps {
+  open: boolean;
+  dropdown: React.ReactNode;
+  preferredDirections?: Direction[];
+  onClose?: () => void;
```

Comment:
it's optional here with no default value, but is required in `DropdownContents`, potential 💥 ?

Reply:
https://github.com/dcos-labs/ui-kit/pull/256/commits/538a723b168be2c487f940b991535c0d0b594cb2